server/src/test/java/org/elasticsearch/indices/state/OpenCloseIndexIT.java

```diff
@@ -317,6 +322,26 @@ public void testOpenWaitingForActiveShardsFailed() throws Exception {
         ensureGreen("test");
     }
 
+    public void testCloseUnassignedIndex() throws Exception {
```

Comment:
Let's try to avoid writing these kind of integration tests that simulate a complicated series of steps to reproduce a corner case. Maybe instead add a unit test to `ActiveShardCountTests`?

Reply:
> Maybe instead add a unit test to `ActiveShardCountTests`?

Ok

> Let's try to avoid writing these kind of integration tests that simulate a complicated series of steps to reproduce a corner case.

Well, I've been surprised that we didn't have any integration test that closes an unassigned index so I added one. On the other hand this won't play well with replicated closed indices so it's better to not add it, I agree.